### DISCUSSION TITLE
Allocation profiling generates no output?

### DISCUSSION THREAD

**Discussion by ksafonov** [async-profiler/async-profiler#400] [Q&A] [answered]
Hi,

Latest version of async profiler generates an empty file in "alloc" mode:
```
$ profilers/async-profiler-2.0-rc-linux-x64/profiler.sh -v
Async-profiler 2.0-rc built on Mar  4 2021
Copyright 2016-2021 Andrei Pangin

$ profilers/async-profiler-2.0-rc-linux-x64/profiler.sh -d 300 -f /tmp/1 -e alloc -o collapsed 14043
Profiling for 300 seconds
Done

$ ll /tmp/1
-rw-rw-r-- 1 ksafonov ksafonov 0 Mar 15 09:39 /tmp/1
```

It works fine when in cpu mode:
```
$ profilers/async-profiler-2.0-rc-linux-x64/profiler.sh -d 300 -f /tmp/1 -e cpu  -o collapsed 14043
Profiling for 300 seconds
Done
$ ll /tmp/1
-rw-rw-r-- 1 ksafonov ksafonov 73299221 Mar 15 13:32 /tmp/1
```

Just to check, previous version works fine in allocations mode for this Java process:

```
$ profilers/async5/profiler.sh -v
Async-profiler 1.5 built on Feb 12 2019
Copyright 2018 Andrei Pangin

$ profilers/async5/profiler.sh -d 300 -f /tmp/1 -e alloc -o collapsed 14043
$ ll /tmp/1
-rw-rw-r-- 1 ksafonov ksafonov 55581080 Mar 15 09:46 /tmp/1
```

**Comment by apangin** [comment, marked as answer]
You have loaded multiple versions of the profiler agent (2.0-rc and 1.5) into the same JVM process. These versions conflict with each other: the older agent has overwritten the signal handler set up by the newer one, so the 2.0-rc agent does not receive proper notifications.

I'll think how to handle conflicts better in future versions, but now you need to restart the JVM process and use just one profiler agent.

**Comment by ksafonov** [comment]
Thanks Andrey! I would ask the profiler to display respective error message then.
